cdc/generation.cc

```diff
@@ -374,4 +376,130 @@ void update_streams_description(
     }
 }
 
+static db_clock::time_point as_timepoint(const utils::UUID& uuid) {
+    return db_clock::time_point{std::chrono::milliseconds(utils::UUID_gen::get_adjusted_timestamp(uuid))};
+}
+
+static future<std::vector<db_clock::time_point>> get_cdc_desc_v1_timestamps(
```

Comment:
This function seems familiar. Is it a copy of another place that does the same thing?

Reply:
No, it's original, the one and only.
What do you mean?

Reply:
I must have seen thing retry loop in other contexts. Maybe it's worth having a util.

Reply:
Maybe, maybe not necessarily
I'd say it's simple enough that an abstraction would only introduce unnecessary complexity with disputable gains (code deduplication is not a goal on its own)